File: backend/models/Genre.go
package models

import "gorm.io/gorm"

type Genre struct {
	gorm.Model
	Tag       string   `gorm:"not null;index"`
	GenreType string   `gorm:"not null" json:"genreType"`
	Movies    []*Movie `gorm:"many2many:movie_genres"`
}

File: prototype/models/Genre.js
import { db as mongoose } from "../lib/db";

const GenreSchema = new mongoose.Schema({
  tag: {
    type: String,
    required: [true, "tag is required"],
    trim: true,
    unique: true,
    index: true,
  },

  genreType: {
    type: String,
    required: [true, "genre type is required"],
    enum: ["music", "movie"],
  },
});

const Genre = mongoose.model("genre", GenreSchema);

export default Genre;
